pandas/tests/api/test_api.py

```diff
@@ -28,7 +28,7 @@ class TestPDApi(Base):
     ignored = ['tests', 'locale', 'conftest']
 
     # top-level sub-packages
-    lib = ['api', 'compat', 'core', 'errors', 'pandas',
+    lib = ['api', 'arrays', 'compat', 'core', 'errors', 'pandas',
```

Comment:
why are we exposing this?

Reply:
If we're doing this, we'll need to put them somewhere in the public API so that we can include them in the docs. https://github.com/pandas-dev/pandas/pull/23581 is collecting them under `.arrays`. (I think @jorisvandenbossche didn't want them in `api.extensions`, since that's for people writing EAs, not just using them).

Reply:
we already have pandas.core.arrays, why add additional exposure here? that seems very very odd

Reply:
I've been assuming that `pandas.core.arrays` and everything under it is private. If we're saying that it's public then I'll want to make some helper methods private.

Reply:
Yes, everything in `pandas.core` should be considered private. So if we want to make the classes public, we need to expose them somewhere else. We could maybe also opt to, for now, not make the classes public and wait until people come up with a use case? (I think the `pd.array` should already be sufficient for most use cases)

Reply:
I think that if users get back an object as part of a normal operation then it's de-facto part of the public API.

Reply:
-1 on expanding pandas namespace like this, no reason this should not be in pandas.core.arrays

Reply:
This is the same as #23581. pandas.core is private, so we shouldn't be adding things from there to the public API